frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerLifeCycle.java

```diff
@@ -380,10 +380,14 @@ public void terminate() {
         @Override
         public void run() {
             try (Scanner scanner = new Scanner(is, StandardCharsets.UTF_8.name())) {
-                while (isRunning.get() && scanner.hasNext()) {
+                while (scanner.hasNextLine()) {
```

Comment:
I assume that hasNextLine() will return True for the input stream as long as the process is alive even though the stream does not contain new input?

Reply:
I believe [hasNextLine()](https://docs.oracle.com/javase/8/docs/api/java/util/Scanner.html#hasNextLine--) will block until the next line is available and return `true` once a new line of input is available. It will return `false` once `EOF` is encountered or underlying stream has been closed.